libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1250,6 +1251,34 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
         break;
     }
 
+    case AUX_FUNC::ARM_EMERGENCY_STOP: {
+        switch (ch_flag) {
+           case AuxSwitchPos::HIGH:
+              SRV_Channels::set_emergency_stop(false);
+              AP::arming().arm(AP_Arming::Method::AUXSWITCH, true);
+              if (hal.util->get_soft_armed()) {
+               // if already armed, log E-stop being de-activated
+                   AP_Logger *logger = AP_Logger::get_singleton();
+                   if (logger && logger->logging_enabled()) {
+                       logger->Write_Event(LogEvent::MOTORS_EMERGENCY_STOP_CLEARED);
+                   }
+               }
+               break;
+           case AuxSwitchPos::MIDDLE:
+               // nothing
+               break;
+           case AuxSwitchPos::LOW:
+               SRV_Channels::set_emergency_stop(true);
+               // log E-stop
+               AP_Logger *logger = AP_Logger::get_singleton();
+               if (logger && logger->logging_enabled()) {
+                   logger->Write_Event(LogEvent::MOTORS_EMERGENCY_STOPPED);
+               }
```

Comment:
Might be worth moving this logging down into servo channels now it is in two places.

Reply:
I can do that

Reply:
done